tools/ImportTool.Shared/SingleFileSpliter.cs

```diff
@@ -0,0 +1,199 @@
+            SplitSizeInBytes = splitSizeInBytes;
+            MaxSpliterCountPerFile = maxSpliterCountPerFile;
+            MaxUploaderCountPerSplitedFile = maxUploaderCountPerSplitedFile;
+            BlockSizeInBytes = blockSizeInBytes;
+        }
+
+        public long SplitSizeInBytes { get; set; }
+
+        public int MaxSpliterCountPerFile { get; set; }
+
+        public int MaxUploaderCountPerSplitedFile { get; set; }
+
+        public long BlockSizeInBytes { get; set; }
+
+        public async Task Split()
+        {
+            var points = GetSplitPoints();
+
+            List<Task> runningTasks = new List<Task>();
+
+            foreach (var (blockMetas, targetCloudBlockBlob) in points)
+            {
+                if (runningTasks.Count >= MaxSpliterCountPerFile)
+                {
+                    var task = await Task.WhenAny(runningTasks.ToArray());
+
+                    await task;
+                    runningTasks.Remove(task);
+                }
+
+                runningTasks.Add(PutNewSplitedFile(blockMetas, targetCloudBlockBlob));
+            }
+
+            await Task.WhenAll(runningTasks);
+        }
+
+        private IEnumerable<Tuple<IEnumerable<BlockMeta>, CloudBlockBlob>> GetSplitPoints()
+        {
+            using (var stream = _sourceBlob.OpenRead())
+            {
+                long totalLength = _sourceBlob.Properties.Length;
+                IList<Tuple<long, long>> points = new List<Tuple<long, long>>();
+                long offset = 0;
+                long length = 0;
+                int index = 0;
+
+                while (true)
+                {
+                    // put all splited files in one hierarchy
+                    string splitedFileName = _sourceBlob.Name.Split('/').Last();
+                    CloudBlockBlob cloudBlockBlob = _sourceBlob.Container.GetBlockBlobReference($"splited/{splitedFileName}_{SplitSizeInBytes}_{index}");
+                    IEnumerable<BlockMeta> blockMetas;
+
+                    length = SplitSizeInBytes;
+
+                    if (stream.Position + length >= totalLength)
+                    {
+                        blockMetas = SplitFileToBlocks(offset, totalLength - offset);
+                        yield return new Tuple<IEnumerable<BlockMeta>, CloudBlockBlob>(blockMetas, cloudBlockBlob);
+                        break;
+                    }
+
+                    stream.Seek(offset + length - 1, SeekOrigin.Begin);
+
+                    // Reads a line. A line is defined as a sequence of characters followed by
+                    // a carriage return ('\r'), a line feed ('\n'), or a carriage return
+                    // immediately followed by a line feed.
+                    int count = 0;
+                    while (true)
+                    {
+                        int c = stream.ReadByte();
+
+                        // \n - UNIX   \r\n - DOS   \r - Mac
+                        if (c == -1 || (c == 10 || c == 13))
```

Comment:
we need to make it clear, -1: end, 10 \r, 13 \n

Reply:
fixed